src/transformers/tokenization_utils_base.py

```diff
@@ -930,28 +936,32 @@ def add_special_tokens(
                     isinstance(t, (str, AddedToken)) for t in value
                 ), f"Tokens {value} for key {key} should all be str or AddedToken instances"
 
+                to_add = []
+                for token in value:
+                    if isinstance(token, (str)):
+                        # for legacy purpose we default to stripping. `test_add_tokens_tokenizer` depends on this
```

Comment:
We don't need to keep things for tests - tests should reflect what behaviour we want. If it changes the tests should change too